src/lib/builders/popover/index.ts

```diff
@@ -0,0 +1,147 @@
+import { derived, writable } from 'svelte/store';
+import {
+	effect,
+	elementDerived,
+	elementMultiDerived,
+	getElementByMeltId,
+	getPlacement,
+	isBrowser,
+	styleToString,
+	sleep,
+	type PositionOptions,
+	createFocusTrap,
+	cleanup,
+	useClickOutside,
+} from '$lib/internal/helpers';
+
+type CreatePopoverArgs = {
+	positioning?: PositionOptions;
+	arrowSize?: number;
+	open?: boolean;
+};
+
+const defaults = {
+	positioning: {
+		placement: 'bottom',
+	},
+	arrowSize: 8,
+	open: false,
+} satisfies CreatePopoverArgs;
+
+export function createPopover(args?: CreatePopoverArgs) {
+	const options = { ...defaults, ...args } as CreatePopoverArgs;
+	const positioning = writable(options.positioning);
+	const arrowSize = writable(options.arrowSize);
+	const open = writable(options.open);
+
+	const activeTrigger = writable<HTMLElement | null>(null);
+
+	const trigger = elementMultiDerived([open], ([$open], createAttach) => {
+		return () => {
+			const attach = createAttach();
+			attach('click', (e) => {
+				e.stopPropagation();
+				const triggerEl = e.currentTarget as HTMLElement;
+
+				open.update((prev) => {
+					const isOpen = !prev;
+					if (isOpen) {
+						activeTrigger.set(triggerEl);
+					} else {
+						activeTrigger.set(null);
+					}
+					return isOpen;
+				});
+			});
+
+			return {
+				role: 'button' as const,
+				'aria-haspopup': 'dialog' as const,
+				'aria-expanded': $open,
+				'data-state': $open ? 'open' : 'closed',
+			};
+		};
+	});
+
+	let popoverCleanup: ((() => void) | undefined)[] = [];
```

Comment:
This is a clever idea, but I think it shows a shortcoming of `elementDerived`. What if we added an addUnsubscriber method to it?

Reply:
I took the liberty of doing that 😄

Reply:
I was going to say that's something that I felt was missing that may end up being useful down the line! Awesome!